src/cyclotom.c

```diff
@@ -854,7 +854,7 @@ static Obj FuncSetCyclotomicsLimit(Obj self, Obj newlimit)
                      CyclotomicsLimit, 0);
     }
 #ifdef SYS_IS_64_BIT
-    if (ulimit > (1L << 32)) {
+    if (ulimit >= (1L << 32)) {
```

Comment:
Could you please also add a test case, in `tst/testinstall/cyclotom.tst` around line 279 would be fitting